### PR TITLE
Update from_numpyro converter to follow schema convention

### PR DESCRIPTION
**Pull request by utkarsh-maheshwari**
## Description
Matched sample stats to ArviZ names in `from_numpyro` converter. Addreses issue #1514

## Checklist

- [x] Follows [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) PR format
- [ ] Code style  correct (follows pylint and black guidelines)
- [ ] Changes are listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased)

### CODE REVIEW

**Comment by utkarsh-maheshwari** [comment]
Please review the changes. As I am a beginner, I am sorry if any conflict arises.

File: arviz/data/io_numpyro.py [hunk 1]
```diff
@@ -132,12 +132,12 @@ def posterior_to_xarray(self):
     @requires("posterior")
     def sample_stats_to_xarray(self):
         """Extract sample_stats from NumPyro posterior."""
-        # match PyMC3 stat names
+        # match arviZ stat names
```
Comment (OriolAbril) [inline arviz/data/io_numpyro.py:135]: I think we can delete the comment now, all libraries will have some kind of renaming to match arviz convention.

**Review by OriolAbril — COMMENTED** [review]
Note that there is an extra sample stat added in line 150:

    data["depth"] = np.log2(value).astype(int) + 1

which should also be updated.

**Comment by OriolAbril** [comment]
Great, do you know jow to rebase? We merged a fix to have ci tests pass again and it is not included in this pr unless we rebase it, this is why tests are failing.

If you don't know yet don't worry, we can guide you

**Comment by utkarsh-maheshwari** [comment]
Oh! I was looking for a solution to these failed checks.
I know how rebasing works. 
Just a small doubt. Do I need to update my forked repository for rebasing? (It is some commits behind the main repository) 
Or can I directly rebase `my-feature`branch with origin?

**Comment by OriolAbril** [comment]
You should rebase with respect to the main repo, not your fork, which is probably called upstream instead of origin (generally the name of the fork). Here is some github guidance on that: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork (I personally prefer using `git rebase upstream/main` in step 5 instead of merge). Once you have rebased your local repo, you can also update your fork, checkout your local main branch, rebase it to upstream/main and then push to your fork, I would recommend doing that, but it is optional, not compulsory.

**Comment by utkarsh-maheshwari** [comment]
Thank you @OriolAbril for pointing out the mistake in my comment. It is more clear now.
Seems like I messed up while rebasing. I attempted to rebase before your comment and probably made a mistake there.
Actually I updated my fork first by creating a pull request and then when I was trying to push from my local branch, I was prompted by an error
image
While resolving this error, I probably messed up the process.
Though I learnt a lot about git. But there is so much more to learn about it.

I want to re-attempt it. May I do this work in a different pull request?

**Comment by OriolAbril** [comment]
There is no need for a new PR. Try running:

    git rebase --interactive HEAD~10

You should get an editor open with the 10 latest commits shown with the message pick before them, edit the file to drop all the undesired commits (all update and merge commits and all commits that are not yours) and save and close the file. There is more on interactive rebase here https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History.

This should modify your local branch to the contributions of this pr only. This is a clean slate. Once there rebase from upstream/main.

After the rebase check that everything is correct, do not push if it's not! Very important,  you may loose changes. You can check the changes with `git diff upstream/main` and push with `git push -f`

**Comment by utkarsh-maheshwari** [comment]
This time I compared `my-feature` branch and `upstream/main` branch to confirm that there is only one file changed that is `io_numpyro.py`. This is the only change that reflected in my last commit. 
What could be probable causes of failed tests?

**Comment by ahartikainen** [comment]
Hi, you should rebase this branch against main

    git fetch --all
    git rebase -i upstream/main
    git push -f

**Comment by utkarsh-maheshwari** [comment]
Hi @ahartikainen,
Thank you for replying. I tried commands you suggested. It made the work tree more clean. I was expecting tests to pass this time.
But instead, this time Benchmarks failed too. Isn't rebasing done right?

**Comment by ahartikainen** [comment]
Errors probably don't have anything to do with your pr. (Some cache problem in CI cc @OriolAbril )

**Comment by utkarsh-maheshwari** [comment]
Can I help in any way? Though I don't know much about CI tests, I can read about them if I am allowed to contribute in CI tests as a beginner.

**Comment by OriolAbril** [comment]
Took a quick look and I don't yet have any idea about how to solve this. I will try to take a better look whenever I have some time but it could be a few days from now.

@utkarsh-maheshwari you can work on ci job definition, however there are some ci related things you can't do such as changing secrets or restarting jobs which is not too important in general. However, I don't really know where you should start reading about this, I myself am not a ci expert either. If you are interested in azure and continuous integration look into that, but it may be more useful to continue with the PR or open other PRs (ie update pyro too in addition to numpyro). It is possible that this is an Azure bug or a cache error and rerunning the jobs 2 days from now "fixes" things by itself. Azure and CI can be quite a mystery 😅

**Comment by OriolAbril** [comment]
@michaelnowotny hi! It looks like there is some issue when running pyjags on Azure. Figured it would be worth asking if you have any idea why and how to fix it?

**Comment by michaelnowotny** [comment]
> @michaelnowotny hi! It looks like there is some issue when running pyjags on Azure. Figured it would be worth asking if you have any idea why and how to fix it? 

Hi @OriolAbril. I haven't used Azure (either with or without pyjags). What I do know is that PyJAGS only works on MacOS or Linux. The package does not install on Windows systems. (It seems to be related to the linking step where the binary jags dll cannot be found. This is something  that I had wanted to take a look. If there is a Python/C++ wizard who works on Windows please step forward!) If the Azure instance is running Windows that could be the cause of the problem. Otherwise, an error Log might help diagnose the issue. In any case, thanks for bringing this to my attention!

**Comment by OriolAbril** [comment]
Azure uses Ubuntu 18 in our case. You can see the log of the failing pipeline here if it helps: https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3849&view=logs&jobId=c4dce305-92e4-5069-101d-6f0a07367dc8&j=c4dce305-92e4-5069-101d-6f0a07367dc8&t=4fccf97c-9d97-5f1a-efbe-aa8458975bbb (you can algo get there by clicking on the ci checks of the PR)

**Comment by utkarsh-maheshwari** [comment]
> Took a quick look and I don't yet have any idea about how to solve this. I will try to take a better look whenever I have some time but it could be a few days from now.
> 
> @utkarsh-maheshwari you can work on ci job definition, however there are some ci related things you can't do such as changing secrets or restarting jobs which is not too important in general. However, I don't really know where you should start reading about this, I myself am not a ci expert either. If you are interested in azure and continuous integration look into that, but it may be more useful to continue with the PR or open other PRs (ie update pyro too in addition to numpyro). It is possible that this is an Azure bug or a cache error and rerunning the jobs 2 days from now "fixes" things by itself. Azure and CI can be quite a mystery sweat_smile

Okay then I'll first open other PR's and try to solve the mystery later.

**Comment by OriolAbril** [comment]
Uoo, turns out numpyro tests are checking for the presence of all sample stats: https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3871&view=logs&j=af9a1d3a-ff8f-5483-6823-86518c773a9b&t=d1f9f1df-bd17-5f75-5a7b-8e272c593e8b&l=62 (this message means that these two `(group, var_name)` combinations were exepected but not found in the resulting inferencedata. You can edit the tests at https://github.com/arviz-devs/arviz/blob/main/arviz/tests/external_tests/test_data_numpyro.py to check for the right names.

**Comment by utkarsh-maheshwari** [comment]
Thanks for helping! @OriolAbril @ahartikainen

**Comment by OriolAbril** [comment]
Thanks for the contribution @utkarsh-maheshwari !

**Pull request merged**